src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.security.SecureRandom;
+
+public class FigureSupplier {
+    private static final int BOUND_OF_RANDOM = 10;
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final SecureRandom random = new SecureRandom();
+
+    public int getRandomNumber() {
+        return random.nextInt(BOUND_OF_RANDOM);
+    }
+
+    public TypeFigure getRandomTypeFigure() {
+        return TypeFigure.values()[random.nextInt(TypeFigure.values().length)];
+    }
+
+    public Figure getRandomFigure() {
+        int randomSide = getRandomNumber();
+        int randomSide1 = getRandomNumber();
+        int randomSide2 = getRandomNumber();
+        int randomRadius = getRandomNumber();
```

Comment:
for `CIRCLE` case `randomSide`, `randomSide1`, `randomSide2` are useless, no need to create them in that step. Just call `getRandomNumber()` directly.